testing/it/it.ui.content/src/content/jcr_root/content/core-components/adaptive-image-servlet/image-with-policy/.content.xml

```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~ Copyright 2022 Adobe
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~-->
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:cq="http://www.day.com/jcr/cq/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0" xmlns:nt="http://www.jcp.org/jcr/nt/1.0"
+    jcr:primaryType="cq:Page">
+    <jcr:content
+        cq:lastModified="{Date}2021-05-12T13:15:57.599+02:00"
+        cq:lastModifiedBy="admin"
+        cq:template="/conf/core-components/settings/wcm/templates/simple-template"
+        jcr:primaryType="cq:PageContent"
+        jcr:title="Simple Page"
+        sling:resourceType="core/wcm/components/page/v2/page">
+        <root
+            jcr:lastModified="{Date}2020-11-27T17:51:51.431+02:00"
+            jcr:lastModifiedBy="admin"
+            jcr:primaryType="nt:unstructured"
+            sling:resourceType="core/wcm/components/container/v1/container"
+            layout="simple">
+            <container
+                jcr:created="{Date}2020-11-27T17:53:54.583+02:00"
+                jcr:createdBy="admin"
+                jcr:lastModified="{Date}2020-11-27T17:54:02.429+02:00"
+                jcr:lastModifiedBy="admin"
+                jcr:primaryType="nt:unstructured"
+                sling:resourceType="core/wcm/components/container/v1/container">
+                <image
+                    jcr:created="{Date}2020-11-27T19:42:34.014+02:00"
+                    jcr:createdBy="admin"
+                    jcr:lastModified="{Date}2020-11-27T19:42:37.640+02:00"
+                    jcr:lastModifiedBy="admin"
+                    jcr:primaryType="nt:unstructured"
+                    sling:resourceType="core/wcm/components/image/v2/image"
+                    fileReference="/content/dam/core-components/Adobe_Systems_logo_and_wordmark.png"/>
```

Comment:
@jckautzmann 
In the `image-with-policy` page, could you please point me where the cq policy is being referred?

I've tried to run this test locally with 
```
mvn verify -Ptest-all -Dgranite.it.author.url=http://localhost:4502
```
but it skipped all the tests.. not sure why
```
[INFO] Running com.adobe.cq.wcm.core.components.it.http.page.v2.PageIT
[main] INFO com.adobe.cq.testing.util.GraniteBackwardsCompatibility - Set sling.it.author.url=http://localhost:4502 from granite.it.author.url
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO org.apache.sling.testing.junit.rules.instance.util.ConfigurationPool - Reading initial configurations from the system properties
[main] INFO org.apache.sling.testing.junit.rules.instance.util.ConfigurationPool - Found 1 instance configuration(s) from the system properties
[main] INFO org.apache.sling.testing.junit.rules.instance.ExistingInstanceStatement - InstanceConfiguration (URL: http://localhost:4502, runmode: author) found for test com.adobe.cq.wcm.core.components.it.http.page.v2.PageIT
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.06 s - in com.adobe.cq.wcm.core.components.it.http.page.v2.PageIT
[INFO] Running com.adobe.cq.wcm.core.components.it.http.page.v3.PageIT
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO org.apache.sling.testing.junit.rules.instance.ExistingInstanceStatement - InstanceConfiguration (URL: http://localhost:4502, runmode: author) found for test com.adobe.cq.wcm.core.components.it.http.page.v3.PageIT
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 s - in com.adobe.cq.wcm.core.components.it.http.page.v3.PageIT
[INFO] Running com.adobe.cq.wcm.core.components.it.http.ContainerIT
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO org.apache.sling.testing.junit.rules.instance.ExistingInstanceStatement - InstanceConfiguration (URL: http://localhost:4502, runmode: author) found for test com.adobe.cq.wcm.core.components.it.http.ContainerIT
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.001 s - in com.adobe.cq.wcm.core.components.it.http.ContainerIT
[INFO] Running com.adobe.cq.wcm.core.components.it.http.SeoIT
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO org.apache.sling.testing.junit.rules.instance.ExistingInstanceStatement - InstanceConfiguration (URL: http://localhost:4502, runmode: author) found for test com.adobe.cq.wcm.core.components.it.http.SeoIT
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0.002 s - in com.adobe.cq.wcm.core.components.it.http.SeoIT
[INFO] Running com.adobe.cq.wcm.core.components.it.http.AdaptiveImageServletIT
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO com.adobe.cq.testing.junit.rules.ConfigurableInstance - Using LoginToken Auth as default. Index lane detection: false
[main] INFO org.apache.sling.testing.junit.rules.instance.ExistingInstanceStatement - InstanceConfiguration (URL: http://localhost:4502, runmode: author) found for test com.adobe.cq.wcm.core.components.it.http.AdaptiveImageServletIT
[WARNING] Tests run: 1, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 0 s - in com.adobe.cq.wcm.core.components.it.http.AdaptiveImageServletIT
```

Reply:
I get a successful test run with the following setup:
- AEM author instance running at http://localhost:4502/
- AEM publish instance running at http://localhost:4503/
- content package `core.wcm.components.it` deployed to the author instance
- command: `mvn verify -Ptest-all -[Dgranite.it](http://dgranite.it/).author.url=http://localhost:4502/ -[Dgranite.it](http://dgranite.it/).publish.url=http://localhost:4503/ -Dit.test=AdaptiveImageServletIT`
result:
```
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.488 s - in com.adobe.cq.wcm.core.components.it.http.AdaptiveImageServletIT
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

The policy for the image referenced in `image-with-policy` page is:
```
cq:policy="core/wcm/components/image/v2/image/policy_1606495136242"
```

Reply:
oh ok. I see that the policy is applied from the template used in `image-with-policy` at `/conf/core-components/settings/wcm/templates/simple-template/policies/jcr:content/root/container/core/wcm/components/image/v2/image`

For running the test, I was not using the publish instance, so that was the problem. 
Now, after running publish instance, I'm getting 
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.628 s <<< FAILURE! - in com.adobe.cq.wcm.core.components.it.http.AdaptiveImageServletIT
[ERROR] testAdaptiveImageServletPolicyDelegate(com.adobe.cq.wcm.core.components.it.http.AdaptiveImageServletIT)  Time elapsed: 0.32 s  <<< ERROR!
org.apache.sling.testing.clients.ClientException: 
Expected HTTP Status: 200 . Instead 404 was returned!
 Response Content:
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html>
    <head><title>404 Not Found</title></head>
    <body>
        <h1>Not Found</h1>
        <p>Cannot serve request to /content/core-components/adaptive-image-servlet/image-without-policy/jcr:content/root/container/image.coreimg.80.500.png/1657895067509.png in 1708</p>

        <h3>Request Progress:</h3>
<pre>
      0 TIMER_START{Request Processing}
      0 COMMENT timer_end format is {&lt;elapsed microseconds&gt;,&lt;timer name&gt;} &lt;optional message&gt;
      2 LOG Method=GET, PathInfo=null
      3 TIMER_START{handleSecurity}
    625 TIMER_END{622,handleSecurity} authenticator org.apache.sling.auth.core.impl.SlingAuthenticator@5a2169c0 returns true
    769 TIMER_START{ResourceResolution}
    839 TIMER_END{69,ResourceResolution} URI=/content/core-components/adaptive-image-servlet/image-without-policy/jcr:content/root/container/image.coreimg.80.500.png/1657895067509.png resolves to Resource=JcrNodeResource, type=core/wcm/components/image/v2/image, superType=null, path=/content/core-components/adaptive-image-servlet/image-without-policy/jcr:content/root/container/image
    843 LOG Resource Path Info: SlingRequestPathInfo: path=&apos;/content/core-components/adaptive-image-servlet/image-without-policy/jcr:content/root/container/image&apos;, selectorString=&apos;coreimg.80.500&apos;, extension=&apos;png&apos;, suffix=&apos;/1657895067509.png&apos;
    843 TIMER_START{ServletResolution}
    844 TIMER_START{resolveServlet(/content/core-components/adaptive-image-servlet/image-without-policy/jcr:content/root/container/image)}
    850 TIMER_END{5,resolveServlet(/content/core-components/adaptive-image-servlet/image-without-policy/jcr:content/root/container/image)} Using servlet 1708
    852 TIMER_END{8,ServletResolution} URI=/content/core-components/adaptive-image-servlet/image-without-policy/jcr:content/root/container/image.coreimg.80.500.png/1657895067509.png handled by Servlet=1708
    853 LOG Applying Requestfilters
    855 LOG Calling filter: com.adobe.granite.resourceresolverhelper.impl.ResourceResolverHelperImpl
    857 LOG Calling filter: org.apache.sling.security.impl.ContentDispositionFilter
    858 LOG Calling filter: com.adobe.granite.csrf.impl.CSRFFilter
    861 LOG Calling filter: org.apache.sling.i18n.impl.I18NFilter
    862 LOG Calling filter: com.adobe.granite.httpcache.impl.InnerCacheFilter
    865 LOG Calling filter: org.apache.sling.rewriter.impl.RewriterFilter
    867 LOG Calling filter: com.adobe.cq.history.impl.HistoryRequestFilter
   1082 LOG Calling filter: com.day.cq.wcm.core.impl.WCMRequestFilter
   1087 LOG Calling filter: com.adobe.cq.wcm.core.components.internal.servlets.CoreFormHandlingServlet
   1089 LOG Calling filter: com.adobe.granite.optout.impl.OptOutFilter
   1092 LOG Calling filter: com.day.cq.wcm.foundation.forms.impl.FormsHandlingServlet
   1096 LOG Calling filter: com.day.cq.dam.core.impl.servlet.DisableLegacyServletFilter
   1099 LOG Calling filter: org.apache.sling.engine.impl.debug.RequestProgressTrackerLogFilter
   1100 LOG Calling filter: com.day.cq.wcm.mobile.core.impl.redirect.RedirectFilter
   1101 LOG Calling filter: com.day.cq.wcm.core.impl.AuthoringUIModeServiceImpl
   1150 LOG Calling filter: com.adobe.granite.rest.assets.impl.AssetContentDispositionFilter
   1152 LOG Calling filter: com.adobe.granite.requests.logging.impl.RequestLoggerImpl
   1157 LOG Calling filter: com.adobe.granite.rest.impl.servlet.ApiResourceFilter
   1227 LOG Calling filter: com.day.cq.dam.core.impl.servlet.ActivityRecordHandler
   1233 LOG Calling filter: com.day.cq.dam.core.impl.assetlinkshare.AdhocAssetShareAuthHandler
   1234 LOG Calling filter: com.day.cq.wcm.core.impl.warp.TimeWarpFilter
   1237 LOG Applying Componentfilters
   1238 LOG Calling filter: com.day.cq.personalization.impl.TargetComponentFilter
   1239 LOG Calling filter: com.day.cq.wcm.core.impl.page.PageLockFilter
   1241 LOG Calling filter: com.day.cq.wcm.core.impl.WCMComponentFilter
   1343 LOG Calling filter: com.adobe.granite.metrics.knownerrors.impl.ErrorLoggingComponentFilter
   1344 LOG Calling filter: com.day.cq.wcm.core.impl.WCMDebugFilter
   1352 TIMER_START{1708#0}
   3157 LOG Applying Error filters
   3159 LOG Calling filter: org.apache.sling.i18n.impl.I18NFilter
   3161 LOG Calling filter: org.apache.sling.rewriter.impl.RewriterFilter
   3165 TIMER_START{handleError:status=404}
   3178 TIMER_END{12,handleError:status=404} Using handler BundledScriptServlet (/libs/sling/servlet/errorhandler/404.jsp)
   3238 LOG Adding bindings took 16 microseconds
   4101 LOG Found processor for post processing ProcessorConfiguration: {contentTypes=[text/html], order=-1, active=true, valid=true, processErrorResponse=true, pipeline=(generator=Config(type=htmlparser, config={}), transformers=(Config(type=linkchecker, config={}), Config(type=mobile, config=JcrPropertyMap [node=Node[NodeDelegate{tree=/libs/cq/config/rewriter/default/transformer-mobile: { jcr:primaryType = nt:unstructured, component-optional = true}}], values={jcr:primaryType=nt:unstructured, component-optional=true}]), Config(type=mobiledebug, config=JcrPropertyMap [node=Node[NodeDelegate{tree=/libs/cq/config/rewriter/default/transformer-mobiledebug: { jcr:primaryType = nt:unstructured, component-optional = true}}], values={jcr:primaryType=nt:unstructured, component-optional=true}]), Config(type=contentsync, config=JcrPropertyMap [node=Node[NodeDelegate{tree=/libs/cq/config/rewriter/default/transformer-contentsync: { jcr:primaryType = nt:unstructured, component-optional = true}}], values={jcr:primaryType=nt:unstructured, component-optional=true}]), serializer=Config(type=htmlwriter, config={}))}
   4178 TIMER_END{4173,Request Processing} Dumping SlingRequestProgressTracker Entries
</pre>


        <hr>
    </body>
</html>
(return code=404)
	at com.adobe.cq.wcm.core.components.it.http.AdaptiveImageServletIT.testAdaptiveImageServletPolicyDelegate(AdaptiveImageServletIT.java:48)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   AdaptiveImageServletIT.testAdaptiveImageServletPolicyDelegate:48 » Client Expe...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
```

But when I'm sending the request manually to http://localhost:8080/content/core-components/adaptive-image-servlet/image-without-policy/jcr%3acontent/root/container/image.coreimg.80.500.png/1606818763926.png?contentPolicyDelegatePath=/content/core-components/adaptive-image-servlet/image-with-policy/jcr:content/root/container/image, its working. Not sure why the test is failing.

Reply:
it could be an issue with the suffix `1657895067509.png` that's used by the AIS as the modified date.

Reply:
there might be a mismatch between this suffix and the time when the asset is deployed.

Reply:
hmm.. I see, the test overall looks good to me, if it's working for you, please go ahead and merge.